packages/arb-validator-core/gobridge/arbRollupWatcher.go

```diff
@@ -40,7 +40,15 @@ func newRollupWatcher(address common.Address, client *goEthdata) (*ethRollupWatc
 func (rw *ethRollupWatcher) GetEvents(ctx context.Context, blockId *common.BlockId) ([]arbbridge.Event, error) {
 	rw.client.goEthMutex.Lock()
 	defer rw.client.goEthMutex.Unlock()
-	return rw.client.rollups[rw.rollupAddress].rollup.events[blockId], nil
+	events := make([]arbbridge.Event, 0, len(rw.client.blockMsgs[blockId].msgs[rw.rollupAddress])+len(rw.client.blockMsgs[blockId].msgs[rw.client.ethAddress]))
+	for _, event := range rw.client.blockMsgs[blockId].msgs[rw.rollupAddress] {
+		events = append(events, event)
+	}
+	for _, event := range rw.client.blockMsgs[blockId].msgs[rw.client.ethAddress] {
+		events = append(events, event)
+	}
+
+	return rw.client.blockMsgs[blockId].msgs[rw.rollupAddress], nil
```

Comment:
It looks like you actually ignore the events slice that you built in previously in this function.

Reply:
fixed